### PR TITLE
checking for empty result set

### PR DESCRIPTION
**Pull request by chrisShick**
The findDecrypted method fails when the result set is empty. This is because of the has() check. I propose a check for empty before trying to decrypt.

### CODE REVIEW

File: src/Model/Behavior/CryptBehavior.php [hunk 1]
```diff
@@ -94,29 +94,32 @@ public function beforeSave(Event $event, EntityInterface $entity)
...
     {
-        $options += ['fields' => []];
-        $mapper = function ($row) use ($options) {
-            $driver = $this->_table->connection()->driver();
-            foreach ($this->config('fields') as $field => $type) {
-                if (($options['fields'] && !in_array($field, (array)$options['fields']))
-                    || !$row->has($field)
-                ) {
-                    continue;
+        if(!$query->isEmpty() && $query->isHydrationEnabled()) {
```
Comment (ADmad) [inline src/Model/Behavior/CryptBehavior.php]: The query isn't supposed to be executed at this point, so this empty check is too early.

Comment (chrisShick) [reply src/Model/Behavior/CryptBehavior.php]: I made changes to this. Let me know if this is a better way to do it.

**Comment by chrisShick** [comment]
Is this PR essentially the same as https://github.com/UseMuffin/Crypt/pull/6

**Comment by ADmad** [comment]
@chrisShick Please try that out and let me know if it fixes your issue too.

**Pull request closed without merging**